File: backend/controllers/problemController.js
const Problem = require('../models/Problem');

const getRandomProblems = async (req, res) => {
    const { subject } = req.params;
    try {
        const problems = await Problem.findOne({ subject });
        if (!problems) return res.status(404).json({ error: "No problems found" });

        // Randomize and limit the result to, say, 5 questions
        const randomProblems = problems.questions.sort(() => 0.5 - Math.random()).slice(0, 5);
        res.json(randomProblems);
    } catch (error) {
        res.status(500).json({ error: 'Server error' });
    }
};

module.exports = { getRandomProblems };

File: backend/app/controllers/problem_controller.py
from flask import jsonify
from app.models.problem_model import Problem

def get_random_problems(subject):
    problems = Problem.get_problems_by_subject(subject)
    if problems:
        return jsonify([{
            'id': p.id,
            'subject': p.subject,
            'question': p.question,
            'options': p.options,
            'correct_answer': p.correct_answer,
            'difficulty': p.difficulty
        } for p in problems]), 200
    return jsonify({"error": "No problems found"}), 404